cmd/snap-bootstrap/cmd_initramfs_mounts_test.go

```diff
@@ -5932,3 +5935,22 @@ func (s *initramfsMountsSuite) TestInitramfsMountsTryRecoveryHealthCheckFails(c
 	c.Check(rebootCalls, Equals, 1)
 	c.Check(s.logs.String(), testutil.Contains, `try recovery system health check failed: mock failure`)
 }
+
+func (s *initramfsMountsSuite) TestMountNonDataPartitionPolls(c *C) {
```

Comment:
I tend to agree with @mardy I would like to see a test where we actually test that the loop in MountNonDataPartitionMatchingKernelDisk does something and measures the transition where the first few iterations the file doesn't exist then it is created in a callback or something, then we see it succeed

Reply:
Sure thing! I added a test that adds the file slightly delayed to ensure that this is tested.